fleet/api/http/metrics.go

```diff
@@ -17,53 +19,61 @@ type metricsMiddleware struct {
 	counter metrics.Counter
 	latency metrics.Histogram
 	svc     fleet.Service
+	auth    mainflux.AuthServiceClient
 }
 
 func (m metricsMiddleware) ViewOwnerByChannelIDInternal(ctx context.Context, channelID string) (agent fleet.Agent, _ error) {
 	defer func(begin time.Time) {
 		labels := []string{
 			"method", "viewOwnerByChannelIDInternal",
 			"owner_id", agent.MFOwnerID,
-			"service", "fleet",
-			"id", agent.MFThingID,
+			"thing_id", agent.MFThingID,
 		}
 
 		m.counter.With(labels...).Add(1)
-		m.latency.With(labels...).Observe(time.Since(begin).Seconds())
+		m.latency.With(labels...).Observe(float64(time.Since(begin).Microseconds()))
 
 	}(time.Now())
 
 	return m.svc.ViewOwnerByChannelIDInternal(ctx, channelID)
 }
 
 func (m metricsMiddleware) ViewAgentBackend(ctx context.Context, token string, name string) (interface{}, error) {
+	ownerID, err := m.identify(token)
```

Comment:
Using this approach, yes.

Reply:
Seems to me we could be using the result from the inner call (ViewAgentBackend in this case) - we just need to figure out how to access it. Maybe it can be stored somewhere